File: python/programmers/Level1/숫자 문자열과 영단어.py
def solution(s):

    num_dic = {
        "zero": "0",
        "one": "1",
        "two": "2",
        "three": "3",
        "four": "4",
        "five": "5",
        "six": "6",
        "seven": "7",
        "eight": "8",
        "nine": "9"
    }

    answer = []
    sub_string = []
    for string in s:
        if string.isdigit():
            answer.append(string)
        else:
            sub_string.append(string)

        key = "".join(sub_string)

        if key in num_dic.keys():
            answer.append(num_dic[key])
            sub_string = []

    return int("".join(answer))


if __name__ == "__main__":
    print(solution("one4seveneight"))
    print(solution("23four5six7"))
    print(solution("2three45sixseven"))
    print(solution("123"))

File: Kotlin/Algorithm/src/숫자 문자열과 영단어.kt
//fun main(args: Array<String>) {
//    println(solution1("one4seveneight"))
//    println(solution1("23four5six7"))
//    println(solution1("2three45sixseven"))
//    println(solution1("123"))
//}

//pravate fun solution1(s: String): Int {
//    val dicNumber: HashMap<String, String>
//    dicNumber = hashMapOf(
//        "one" to "1",
//        "two" to "2",
//        "three" to "3",
//        "four" to "4",
//        "five" to "5",
//        "six" to "6",
//        "seven" to "7",
//        "eight" to "8",
//        "nine" to "9",
//        "zero" to "0"
//    )
//    var answer = ""
//    var stringNumber: String = ""
//    for (str in s) {
//        if (str.isDigit()) answer += str
//        else {
//            stringNumber += str
//        }
//        if (dicNumber.containsKey(stringNumber)) {
//            answer += dicNumber[stringNumber]
//            stringNumber = ""
//        }
//    }
//    return answer.toInt()
//}